<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../jacoco-resources/report.gif" type="image/gif"/><title>GenomicLocationValidator.java</title><link rel="stylesheet" href="../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">Genome Nexus Web App</a> &gt; <a href="index.source.html" class="el_package">org.cbioportal.genome_nexus.web.validation</a> &gt; <span class="el_source">GenomicLocationValidator.java</span></div><h1>GenomicLocationValidator.java</h1><pre class="source lang-java linenums">package org.cbioportal.genome_nexus.web.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import org.cbioportal.genome_nexus.web.validation.ValidGenomicLocation;

<span class="nc" id="L7">public class GenomicLocationValidator implements ConstraintValidator&lt;ValidGenomicLocation, String&gt; {</span>
    
    public void initialize(ValidGenomicLocation constraintAnnotation) {
<span class="nc" id="L10">    }</span>

    public boolean isValid(String genomicLocation, ConstraintValidatorContext context) 
    {
<span class="nc" id="L14">        String[] genomicLocationSplit = genomicLocation.split(&quot;,&quot;);</span>
<span class="nc bnc" id="L15" title="All 2 branches missed.">        if (genomicLocationSplit.length &lt; 5) {</span>
<span class="nc" id="L16">            return false;</span>
        }
        else {
<span class="nc" id="L19">            boolean result = true;</span>
            // chr [1-24,X,Y,MT]
<span class="nc bnc" id="L21" title="All 6 branches missed.">            result &amp;= genomicLocationSplit[0].matches(&quot;X&quot;) || genomicLocationSplit[0].matches(&quot;Y&quot;) || genomicLocationSplit[0].matches(&quot;MT&quot;) ||</span>
<span class="nc bnc" id="L22" title="All 6 branches missed.">            genomicLocationSplit[0].matches(&quot;[0-9]+&quot;) &amp;&amp; Integer.valueOf(genomicLocationSplit[0]) &gt;= 1 &amp;&amp; Integer.valueOf(genomicLocationSplit[0]) &lt;= 24;</span>
            // start &amp; end should be positive integer
<span class="nc bnc" id="L24" title="All 4 branches missed.">            result &amp;= Integer.valueOf(genomicLocationSplit[1]) &gt;= 0 &amp;&amp; Integer.valueOf(genomicLocationSplit[2]) &gt;= 0;</span>
            // variant &amp; ref allele should only contain T,C,G,A,-
<span class="nc bnc" id="L26" title="All 4 branches missed.">            result &amp;= genomicLocationSplit[3].matches(&quot;[TCGA-]*&quot;) &amp;&amp; genomicLocationSplit[4].matches(&quot;[TCGA-]*&quot;);</span>
<span class="nc" id="L27">            return result;</span>
        }
    }
}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.8.7.202105040129</span></div></body></html>